### PR TITLE
Make JRuby compatible, closes #3

### LINKED ISSUE #3: Ensure JRuby compatability

### PR DESCRIPTION
**Pull request by orischwartz**

### CODE REVIEW

**Comment by paultag** [comment]
Will merge when green :+1:

**Comment by paultag** [comment]
```
Installing libv8 3.16.14.13 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
    /home/travis/.rvm/rubies/jruby-9.0.3.0/bin/jruby -r ./siteconf20160210-6293-1drbf5j.rb extconf.rb
jruby: warning: unknown property jruby.cext.enabled
NotImplementedError: C extensions are not supported
    <top> at /home/travis/.rvm/rubies/jruby-9.0.3.0/lib/ruby/stdlib/mkmf.rb:1
  require at org/jruby/RubyKernel.java:939
   (root) at /home/travis/.rvm/rubies/jruby-9.0.3.0/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1
    <top> at /home/travis/.rvm/rubies/jruby-9.0.3.0/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:54
extconf failed, exit code 1
Gem files will remain installed in /home/travis/build/department-of-veterans-affairs/caseflow-certification/vendor/bundle/jruby/2.2.0/gems/libv8-3.16.14.13 for inspection.
Results logged to /home/travis/build/department-of-veterans-affairs/caseflow-certification/vendor/bundle/jruby/2.2.0/extensions/universal-java-1.8/2.2.0/libv8-3.16.14.13/gem_make.out
```

**Comment by paultag** [comment]
@shanear seems like we can't use `libv8` with jruby, is that part of the not-pally accessibility stuff?

**Comment by orischwartz** [comment]
@shanear I will update this so we delay the switch to JRuby as long as possible. We'll still need the other changes here.

File: lib/tasks/lint.rake [hunk 1]
```diff
@@ -10,8 +10,13 @@ task :lint do
...
 
-  puts "\nrunning jshint..."
-  jshint_result = ShellCommand.run("rake jshint")
+  if defined? JRUBY_VERSION
+    # fake a passing result
+    jshint_result = true
+  else
+    puts "\nrunning jshint..."
+    jshint_result = ShellCommand.run("rake jshint")
+  end
```
Comment (paultag) [inline lib/tasks/lint.rake]: Can we just remove jshint from the `lint.rake` file and remove it from the conditionals below?

Comment (paultag) [inline lib/tasks/lint.rake]: Basically, I don't love the `jshint_result = true` on jruby - if we want to do `jshint` too, we could run it below and bail if it fails before checking the other two.

Comment (orischwartz) [inline lib/tasks/lint.rake]: @paultag  Yeah me either....I was trying to make this change in a way that would be compatible with roadrunner so you can start MRI Ruby projects with all linters/features and JRuby projects with a working subset.

If you prefer I also have a version of these rake scripts that don't shell out and instead invoke the rake tasks. Its more portable (no requirements that user have anything on `PATH`) but it's also not as direct and readable as shelling out.

e.g., here's how this file would look:

```Ruby
require "open3"
require "rainbow"
require "scss_lint/rake_task"
require "rubocop/rake_task"

desc "shortcut to run all linting tools, at the same time."
task :lint do
  run_lint_task "scss-lint" do
    SCSSLint::RakeTask.new(:scss_lint) do |t|
      t.args = ["--color"]

      # by default, will lint entire application by specifying '.'; instead, defer to files
      # in .scss-lint.yml
      t.files = []
    end
    Rake::Task[:scss_lint].execute
  end

  run_lint_task "rubocop" do
    RuboCop::RakeTask.new(:rubo_cop)
    Rake::Task[:rubo_cop].execute
  end

  unless defined? JRUBY_VERSION
    require "jshint"
    run_lint_task "jshint" do
      Jshint::Cli.run
    end
  end

  puts "\n"
  puts Rainbow("Passed. Everything looks stylish!").green
end

def run_lint_task(name, &task)
  puts "running #{name}..."
  begin
    yield task
  rescue SystemExit
    puts Rainbow("#{name} failed. Linting issues were found.").red
    exit!(1)
  end
end
```

**Comment by orischwartz** [comment]
Reverted `.ruby-version` `.travis.yml` to delay the use of JRuby as long as possible; we should periodically run against JRuby to make sure we haven't introduced any incompatibilities.

**Comment by shanear** [comment]
@orischwartz i like this.

**Pull request merged**